### PR TITLE
Pass `verbose` to unpack strategies.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/download_strategy.rb [hunk 1]
```diff
@@ -330,7 +330,7 @@ def _fetch
 class NoUnzipCurlDownloadStrategy < CurlDownloadStrategy
   def stage
     UnpackStrategy::Uncompressed.new(cached_location)
-                                .extract(basename: basename_without_params)
+                                .extract(basename: basename_without_params, verbose: ARGV.verbose? && !shutup)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/download_strategy.rb]: This line is a bit long. `verbose` could be extracted to be a variable, perhaps?

File: Library/Homebrew/system_command.rb [hunk 2]
```diff
@@ -46,13 +47,14 @@ def run!
     result
   end
 
-  def initialize(executable, args: [], sudo: false, input: [], print_stdout: false, print_stderr: true, must_succeed: false, env: {}, **options)
+  def initialize(executable, args: [], sudo: false, input: [], print_stdout: false, print_stderr: true, verbose: false, must_succeed: false, env: {}, **options)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/system_command.rb]: Can this line be shrunk so it doesn't wrap on the GitHub UI (120 chars, iirc).

File: Library/Homebrew/unpack_strategy/air.rb [hunk 3]
```diff
@@ -13,10 +13,9 @@ def dependencies
     private
 
     def extract_to_dir(unpack_dir, basename:, verbose:)
-      system_command!(
-        "/Applications/Utilities/Adobe AIR Application Installer.app/Contents/MacOS/Adobe AIR Application Installer",
-        args: ["-silent", "-location", unpack_dir, path],
-      )
+      system_command! "/Applications/Utilities/Adobe AIR Application Installer.app/Contents/MacOS/Adobe AIR Application Installer",
```
Comment (MikeMcQuaid) [inline Library/Homebrew/unpack_strategy/air.rb]: Can this line be shrunk so it doesn't wrap on the GitHub UI (120 chars, iirc).

**Comment by chdiza** [comment]
I think something that made unpacking verbose was done before, and got reverted because it was irritating.  I'm too lazy to go look through the legacy repo, but I think @ilovezfs might have been the one who reverted it.

**Comment by ilovezfs** [comment]
Yeah dumping the list of every file in a source archive is not exactly helpful.

**Comment by MikeMcQuaid** [comment]
> Yeah dumping the list of every file in a source archive is not exactly helpful.

Does this actually do that or does it just output the command names that are used?

**Comment by reitermarkus** [comment]
I think there are two separate questions here:

 - Should `--debug` imply `--verbose`?
 - Should every command be shown when `--verbose`, or just the ones we specifically pass `verbose` to?

**Comment by MikeMcQuaid** [comment]
> Should every command be shown when --verbose/--debug, or just the ones we specifically pass verbose to?

To be more explicit here: by "every command" do you mean "the executed command and arguments" or "the output from the command" (or both/neither)?

**Comment by reitermarkus** [comment]
I mean “the executed command and arguments”.

**Comment by MikeMcQuaid** [comment]
Cool. I think it should be output by both debug and verbose with perhaps output also being output by verbose.

**Comment by reitermarkus** [comment]
> I think it should be output by both debug and verbose

Good, this is the current state:

```ruby
verbose? || ARGV.debug?
```

**Pull request merged**